src/txmempool.h

```diff
@@ -693,8 +717,19 @@ class CTxMemPool
         return (mapTx.count(hash) != 0);
     }
 
+    bool wtxid_exists(const uint256& hash) const
```

Comment:
I think this is just a style choice, trying to balance different tradeoffs:
- if you change a function's arguments, then you have to change all call sites, unless you use an optional argument
- changing lots of call sites is not great as it adds noise to PR review
- optional arguments are not great as they can make it harder to maintain code and do review in the future

But sometimes optional arguments are the easier way to go, to avoid having to embed hairy logic at a bunch of call sites to decide which function to call, which can also harm readability.  So I think it's just a balance, and I tried to strike the balance I thought made sense when I was writing all this which wasn't always the same for every example, but maybe I could have done better.  

Anyway at this point I'd prefer not to make stylistic changes to this PR.